torch_geometric/nn/conv/point_transformer_conv.py

```diff
@@ -0,0 +1,144 @@
+
+from ..inits import reset
+
+
+class PointTransformerConv(MessagePassing):
+    r"""The PointTransformer layer from the `"Point Transformer"
+    <https://arxiv.org/abs/2012.09164>` paper
+
+    .. math::
+        \mathbf{x}^{\prime}_i =  \sum_{j \in
+        \mathcal{N}(i) \cup \{ i \}} \rho \left( \gamma \left( \phi
+        \left( \mathbf{x}_i \right) - \psi \left( \mathbf{x}_j \right) \right)
+        + \delta  \right) \odot \left( \alpha \left(  \mathbf{x}_j \right)
+        + \delta \right) ,
+
+    where :math:`\gamma` and :math:`\delta = \theta \left( \mathbf{p}_j -
+    \mathbf{p}_i \right)` denote neural networks, *.i.e.* MLPs, that
+    respectively map transformed node features and relative spatial
+    coordinates. :math:`\mathbf{P} \in \mathbb{R}^{N \times D}`
+    defines the position of each point.
+
+    Args:
+        in_channels (int): Size of each input sample features
+        out_channels (int) : size of each output sample features
+        local_nn : (torch.nn.Module, optional): A neural network
+            :math:`\delta = \theta \left( \mathbf{p}_j -
+            \mathbf{p}_i \right)` which maps relative spatial coordinates
+            :obj:`pos_j - pos_i` of shape
+            :obj:`[-1, 3]` to shape
+            :obj:`[-1, out_channels]`, *e.g.*, defined by
+            :class:`torch.nn.Sequential`. In case :obj:`None` is given, is
+            initialized with :obj:`Sequential(\n'
+            '    (0): Linear(in_features=3, out_features=64, bias=True)\n'
+            '    (1): ReLU()\n'
+            '    (2): Linear(in_features=64, out_features=32, bias=True)\n'
+            '  )\n'` (default: :obj:`None`)
+        global_nn : (torch.nn.Module, optional): A neural network
+            :math:`\gamma` which maps transformed node features of shape
+            :obj:`[-1, out_channels]` to shape :obj:`[-1, out_channels]`,
+            *e.g.*, defined by :class:`torch.nn.Sequential`. In case
+            :obj:`None` is given, is initialized with :obj:`Sequential(\n'
+            '  (0): Linear(in_features=32, out_features=64, bias=True)\n'
+            '  (1): ReLU()\n'
+            '  (2): Linear(in_features=64, out_features=32, bias=True)\n'
+            ')'` (default: :obj:`None`)
+        add_self_loops (bool, optional) : If set to :obj:`False`, will not add
+            self-loops to the input graph. (default: :obj:`True`)
+        pos_mlp_channels (int, optional) : size of intermediary channels in
+            points relative spatial coordinates computation. Has no effect in
+            case local_nn is not None (default: :obj:`64`)
+        attn_mlp_channels (int, optional) : factor size of intermediary
+            channels in transformed features mlp. Has no effect in
+            case global_nn is not None (default: :obj:`2`)
+        **kwargs (optional): Additional arguments of
+            :class:`torch_geometric.nn.conv.MessagePassing`.
+    """
+
+    def __init__(self,
+                 in_channels,
```

Comment:
Might have to add some code to handle the case where `in_channels` is a tuple of `int`. Refer to [tranformer_conv](https://github.com/pyg-team/pytorch_geometric/blob/b3618d2dc57ad31a420e400ee83e42a5200037e4/torch_geometric/nn/conv/transformer_conv.py#L114).

Reply:
Added. However I do not really understand how it can be used so I hope I did it right